zerver/worker/queue_processors.py

```diff
@@ -115,25 +115,82 @@ def on_failure(event: Dict[str, Any]) -> None:
 
 class QueueProcessingWorker(ABC):
     queue_name = None  # type: str
+    CONSUME_ITERATIONS_BEFORE_UPDATE_STATS_NUM = 50
 
     def __init__(self) -> None:
         self.q = None  # type: SimpleQueueClient
         if self.queue_name is None:
             raise WorkerDeclarationException("Queue worker declared without queue_name")
+        self.initialize_statistics()
+
+    def initialize_statistics(self) -> None:
+        self.queue_last_emptied_timestamp = time.time()
+        self.consumed_since_last_emptied = 0
+        self.recent_consume_times = deque(maxlen=50)  # type: MutableSequence[Tuple[int, float]]
+        self.consume_interation_counter = 0
+
+        self.update_statistics(0)
```

Comment:
What's the thinking behind having this log on startup?  I think that's probably appropriate; and I guess code accessing this has its own ability to get the queue size, so the fact that the `0` here is a lie isn't a problem?  

May be worth a comment.

Reply:
The queue checkers checks *active* queues, that is those with a registered consumer - so I want to update_statistics on startups to avoid the annoying event where the checker run into a stale `.stats` file - so I want the queue worker to write a reasonable initial value on startup. I moved this from `__init__` to `start()` though, because that seems like a more appropriate place for this purpose.